class.json-api.php

```diff
@@ -95,7 +95,10 @@ function setup_inputs( $method = null, $url = null, $post_body = null ) {
 		}
 
 		$parsed     = parse_url( $this->url );
-		$this->path = $parsed['path'];
+		$this->path = '';
+		if ( array_key_exists( 'path', $parsed ) ) {
```

Comment:
How about checking for the actual value instead?

```suggestion
		if ( ! empty( $parsed['path'] ) ) {
```

Reply:
Nice! I wouldn't expect that it will work with undefined indexes :) Also, do you think the default value is needed here?

Reply:
> I wouldn't expect that it will work with undefined indexes :)

`empty()` will return true if a variable is undefined, null, false, 0 or an empty string, so that should cover our needs here.

> do you think the default value is needed here?

We may not actually need it since that's already the default that is set for the member variable.

Reply:
Thanks! Feedback addressed.